solidity/test/bridge/Bridge.test.ts

```diff
@@ -2233,229 +2252,1840 @@ describe("Bridge", () => {
+                        })
+
+                        it("should revert", async () => {
+                          await expect(outcome).to.be.revertedWith(
+                            "Output value is not within the acceptable range of the pending request"
+                          )
+                        })
+                      }
+                    )
+
+                    context(
+                      "when output vector contains a reported timed out requested redemption with wrong amount",
+                      () => {
+                        const data: RedemptionTestData = JSON.parse(
+                          JSON.stringify(MultiplePendingRequestedRedemptions)
+                        )
+
+                        let outcome: Promise<RedemptionScenarioOutcome>
+
+                        before(async () => {
+                          await createSnapshot()
+
+                          // Alter the last redemption requests in the test
+                          // data and set such an amount that will cause
+                          // the Bitcoin redemption transaction to be deemed
+                          // as invalid due to a wrong amount. The corresponding
+                          // transaction output has the value of 191169 so to
+                          // make this test scenario happen, the request amount
+                          // must be way different (lesser or greater) than the
+                          // output value.
+                          data.redemptionRequests[4].amount = 100000
+
+                          // Before submitting the redemption proof, wait
+                          // an amount of time that will make the last request
+                          // timed out and then report the timeout.
+                          const beforeProofActions = async () => {
+                            await increaseTime(await bridge.redemptionTimeout())
+                            await bridge.notifyRedemptionTimeout(
+                              data.wallet.pubKeyHash,
+                              data.redemptionRequests[4].redeemerOutputScript
+                            )
+                          }
+
+                          outcome = runRedemptionScenario(
+                            data,
+                            beforeProofActions
+                          )
+                        })
+
+                        after(async () => {
+                          await restoreSnapshot()
+                        })
+
+                        it("should revert", async () => {
+                          await expect(outcome).to.be.revertedWith(
+                            "Output value is not within the acceptable range of the timed out request"
+                          )
+                        })
+                      }
+                    )
+
+                    context(
+                      "when output vector contains a non-zero P2SH change output",
```

Comment:
We expect it to revert because P2SH-type main UTXO is not allowed and only P2PKH is valid, correct?

https://developer.bitcoin.org/devguide/transactions.html says:

> As of Bitcoin Core 0.9, the standard pubkey script types are:
> - Pay To Public Key Hash (P2PKH)
> - Pay To Script Hash (P2SH)
> - Multisig
> - Pubkey
> - Null Data

Should we have similar cases for the last three? What about witness?